types/hapi/index.d.ts

```diff
@@ -3481,6 +3481,7 @@ export class Server extends Podium {
     decorate(type: 'toolkit', property: string, method: DecorationMethod<ResponseToolkit>, options?: {apply?: boolean, extend?: boolean}): void;
     decorate(type: 'server', property: string, method: (existing: ((...args: any[]) => any)) => DecorationMethod<Server>, options: {apply?: boolean, extend: true}): void;
     decorate(type: 'server', property: string, method: DecorationMethod<Server>, options?: {apply?: boolean, extend?: boolean}): void;
+    decorate(type: 'request'|'server', property: string, method: any, options?: {apply?: boolean, extend?: boolean}): void;
```

Comment:
This breaks type safety for this function.

Reply:
This was the goal of the change. According to the Hapi source code, you can pass any method or any property when decorating the server or request properties.

With the current types, the signature limited it to some specific types, where `this` was bound to a Request or Server object. As I was converting code from JS to TS, I was getting errors because the signature of objects I was using as decorators (and that had been working fine) didn't match.

Reply:
Perhaps using something like `Exclude<any, Function>` would be better so this doesn't ruin type safety.